### ISSUE TITLE
Different loss term compared to paper

### ISSUE THREAD

**Issue by WhoDunnett** [wanghangpsu/MMAC#1]
Hi MMAC team,

Thank you for making your code available on GitHub for others to explore. While reviewing your code I noticed that the loss defined in your paper differs from the one implemented. Specifically, on line 212 of the mmac.py file, the total loss is calculated as 

loss = loss1 - c * loss3 

However, in the paper, I interpreted this as 

loss = loss1 + c * loss3

I'm sorry if I'm missing something or misinterpreting the paper's notation or code itself. 

Thanks

**Comment by wanghangpsu** [comment]
loss 3 in the code is the negative margin,  So we use the loss=loss - c loss3, which is loss = loss + c margin

**Comment by WhoDunnett** [comment]
Thanks for the clarification! Out of interest, do you have any plans to implement your approach within the BackdoorBench framework? @wanghangpsu

**Comment by wanghangpsu** [comment]
Sorry I am no-longer working on this area, and I might not have time to do so.

**Comment by WhoDunnett** [comment]
No worries. If I were to implement this within that framework, would you or someone still working on this project be able to "validate" my implementation?
